messages/src/decorators/attachment.rs

```diff
@@ -58,6 +59,22 @@ pub enum AttachmentType {
     Links(Vec<Url>),
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, TypedBuilder)]
+#[serde(rename_all = "snake_case")]
+pub struct AttachmentFormatSpecifier<F> {
+    pub attach_id: String,
+    pub format: MaybeKnown<F>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, TypedBuilder)]
+#[serde(rename_all = "snake_case")]
+pub struct OptionalIdAttachmentFormatSpecifier<F> {
+    #[builder(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub attach_id: Option<String>,
+    pub format: MaybeKnown<F>,
+}
+
```

Comment:
I assume these were placed here because they're shared across the V2 protocols, but IMO they don't really belong next to the decorator as they're protocol specific. 

Maybe it's more suitable to create some sort of `common` module in the `protocols` one to share common stuff across protocols?

Reply:
ah yep. agreed, have addressed this in my latest commit. If you're happy with the changes, and the CI is still green, pls merge 👌